internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -729,6 +731,25 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 	return reconcileComplete{dirty: true, stop: true}
 }
 
+//
+//  In assisted installer, UserManagedNetworking implicates none platform.  This flag is part of AgentClusterInstall spec.
+//
+func (r *BMACReconciler) isNonePlatform(ctx context.Context, cd *hivev1.ClusterDeployment) (isNone, requeue bool, err error) {
+	if cd.Spec.ClusterInstallRef == nil {
+		return false, false, errors.Errorf("Cluster Install Reference is null for cluster deployment ns=%s name=%s", cd.Namespace, cd.Name)
```

Comment:
I understand that technically this error and the one in L748 (the Get error) are not the same but, logically (for the flow), they mean pretty much the same. We don't have an ACI that we can use to validate if this is None platform or not.

Regardless of None platform, we always need an ACI when deploying a day2 cluster using the KubeAPI flow so, if we hit this case, then there's a bigger issue and we should not proceed with the current reconciliation.

Unless I'm missing something, of course :)

Reply:
OK.  We will always return complete (although it is a change of flow for BM but it is unexpected)